interceptors/DataApiInterceptors.cfc

```diff
@@ -165,4 +166,21 @@ component extends="coldbox.system.Interceptor" {
 
 		dataApiQueueService.queueInsert( argumentCollection=interceptData );
 	}
-}
\ No newline at end of file
+
+
+
+	public void function onCreateSelectDataCacheKey( event, interceptData ) {
```

Comment:
This feels dangerous / misplaced. This interception point could easily be fired off 1000's of times in any given request (i.e. some bad uncached admin request with lots of little queries). From my understanding, you want to use a 'processed' cache of the select data results, only when the select data result hasn't changed.

I think a more accurate / clean approach here would be:

1. Create a dedicated API result cache
2. Do your selectData as normal
3. Generate a cache key that is API Endpoint + a hash of the db query result

Then just use your dedicated API cache with this cache key. If the data changes in the DB, that cache entry will no longer get looked up.

Does that make sense? Here is an example of an extension registering its own cachebox cache using `afterConfigurationLoad` coldbox interception point: https://github.com/pixl8/preside-ext-s3-storage-provider/blob/stable/interceptors/S3StorageProviderInterceptors.cfc

Reply:
Hi @DominicWatson !

I understand you.

The answer to the question **2 Do your selectData as normal** Is the reason I add the interceptor **onCreateSelectDataCacheKey** To check if the data from the selectData has been cached or not.

- If it was cached then get the cache specific for api result
- If not, processed and save cached

What do you propose to do this without use the intercelpor **onCreateSelectDataCacheKey** ?

It is true **onCreateSelectDataCacheKey** is fired a lot of times but it is also true I did not do any important action there more than set true or false some RequestContext

Reply:
> To check if the data from the selectData has been cached or not.

This does not essentially matter. All that matters is that you have a version in cache that matches the result of the query. If the query is cached _and_ the API result is cached then super-great. If the query is _not_ cached but the result has not changed and we still have an API result cache, then we can still use this API result cache.